beacon_node/store/src/hot_cold_store.rs

```diff
@@ -40,7 +40,10 @@ use std::sync::Arc;
 use std::time::Duration;
 use types::*;
 
-/// On-disk database that stores finalized states efficiently.
+/// The number of replayed states to be cached.
+const STATE_CACHE_SIZE: usize = 32;
```

Comment:
32 is probably too high as a default, and we should also make this configurable.

A CLI param called `--historic-state-cache-size` would be ideal. Have a look at `beacon_node/store/src/config.rs`, and the CLI stuff in `beacon_node/src/cli.rs`. You can copy the logic for the existing `--block-cache-size` flag.

I think the default should probably be 1, to minimize memory usage as much as possible. Users who are interested in historic states can set their own cache size based on their memory budget.